app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdate.kt

```diff
@@ -31,21 +32,27 @@ class TeleconsultPrescriptionUpdate : Update<TeleconsultPrescriptionModel, Telec
       model: TeleconsultPrescriptionModel,
       event: DataForNextClickLoaded
   ): Next<TeleconsultPrescriptionModel, TeleconsultPrescriptionEffect> {
-    return if (event.hasSignatureBitmap) {
-      dispatch(
-          SaveMedicalRegistrationId(medicalRegistrationId = event.medicalRegistrationId),
-          UpdateTeleconsultRecordMedicalRegistrationId(
-              teleconsultRecordId = model.teleconsultRecordId,
-              medicalRegistrationId = event.medicalRegistrationId
-          ),
-          AddTeleconsultIdToPrescribedDrugs(
-              patientUuid = model.patientUuid,
-              teleconsultRecordId = model.teleconsultRecordId,
-              medicalInstructions = event.medicalInstructions
-          )
-      )
-    } else {
-      dispatch(ShowSignatureRequiredError)
+    return when {
+      event.hasSignatureBitmap.not() -> {
+        dispatch(ShowSignatureRequiredError)
+      }
+      event.hasMedicines.not() -> {
```

Comment:
Error scenario : 
Step 1: There's a possibility where the medical officer has not added both Signature and Medicines. In this case, when I click on `Next` button, only signature error is enabled.
Step 2: I add the signature then click on `Next`, we get the medicine error. 

Expected behaviour: On click of `Next`, we should be getting both the errors together if empty.

Reply:
Ah good catch, will add this.